packages/grafana-ui/src/components/Forms/Checkbox.tsx

```diff
@@ -44,34 +44,20 @@ export const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
 
 export const getCheckboxStyles = stylesFactory((theme: GrafanaTheme2) => {
   const labelStyles = getLabelStyles(theme);
-  const checkboxSize = '16px';
+  const checkboxSize = 2;
+  const labelPadding = 1;
+
   return {
-    label: cx(
-      labelStyles.label,
-      css`
-        padding-left: ${theme.spacing(1)};
-        white-space: nowrap;
-        cursor: pointer;
-      `
-    ),
-    description: cx(
-      labelStyles.description,
-      css`
-        line-height: ${theme.typography.bodySmall.lineHeight};
-        padding-left: ${theme.spacing(1)};
-      `
-    ),
     wrapper: css`
       position: relative;
-      padding-left: ${checkboxSize};
       vertical-align: middle;
-      min-height: ${theme.spacing(3)};
+      font-size: 0;
     `,
     input: css`
       position: absolute;
```

Comment:
```suggestion
      position: relative;
```
This seems to fix the issue on manage dashboards page, but breaks the layout on storybook. Can't figure out why it doesn't set the checked state with absolute position on that page. Maybe we'd just keep the custom styles for manage dashboards page 🤔

Reply:
I _think_ the problem maybe is related to `onClick` being used rather than `onChange` - the div wrapping the checkbox registers a click and toggles state in ManageDashboards, but the checkbox itself doesn't receive the click so it doesn't switch to checked. 

I had a brief play to try and switch it over to use `onChange` (which is better for a few reasons) and i think I observed the same problem, but it's definitely possible I didn't dig far enough though. Thinking about it now, perhaps i left in a preventDefault to stopPropagation or something?

Reply:
It does get the checked value though (I also tied `checked` instead of `value`, same effect). plus I don't see how an event handler would influence the styling this way.

Reply:
Yeah, that was also my other confusion - if checked is still going through, why doesn't it switch to the right visual style? 

I might take another stab later this week at understanding this...

Reply:
Is it possible that there's some state changes happening outside of react? Like an angular digest that's not happening? Feel like I'm grasping at straws :P

Reply:
Don't think that'd make any difference with either absolutely or relatively positioned element.

Reply:
I fixed this!

I don't totally understand exactly what was going on, but calling `.preventDefault()` on the event that would check the checkbox would... prevent the checkbox from becoming checked, even though we set state to check it. Here's an extremely minimal reproduction https://codesandbox.io/s/checkbox-repro-o8kos?file=/src/App.js